attestation/check-registration.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# This file is part of Canonical's TDX repository which includes tools
+# to setup and configure a confidential computing environment
+# based on Intel TDX technology.
+# See the LICENSE file in the repository for the license text.
+
+# Copyright 2024 Canonical Ltd.
+# SPDX-License-Identifier: GPL-3.0-only
+
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 3,
+# as published by the Free Software Foundation.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranties
+# of MERCHANTABILITY, SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+
+if [ "$EUID" -ne 0 ]
+  then echo "Please run as root"
+  exit
+fi
+
+# check platform registration
+
+set -e
+
+REGISTRATION_SUCCESS=1
+
+check_mpa_status() {
+    # Run command: mpa_manage -get_last_registration_error_code
+    # If the platform has been registered successfully, the command will output:
+    # Last reported registration error code: 0
+    if mpa_manage -get_last_registration_error_code | grep "error code: 0" 2>&1 > /dev/null
+    then
+	REGISTRATION_SUCCESS=0
+    fi
+}
+
+# check if MPA is used for platform registration
+if systemctl is-enabled mpa_registration_tool.service  2>&1 > /dev/null
+then
+    if command -v mpa_manage 2>&1 > /dev/null
```

Comment:
I assume `mpa_manage` is not installed by default - does this come with a package from the attestation PPA or somewhere else? I'm wondering if it's worthwhile to add an else that logs an error rather than just silently skipping `check_mpa_status`.

Reply:
```
$ dpkg -S /usr/bin/mpa_manage
sgx-ra-service: /usr/bin/mpa_manage
```

Agree for the log, let me do that

Reply:
Done : ac138eb2cb521a414d23b294146a51da6fac35a0

Reply:
@hector-cao In the echo command you added I think you forgot "not", "mpa_manage is not available, ..."

Reply:
Oups, thanks for the catch, fixed : c8b0152811c36b8b8b00228997607c7606f8e9c4